cli/wallet.go

```diff
@@ -235,3 +237,105 @@ var walletImport = &cli.Command{
 		return nil
 	},
 }
+
+var walletSign = &cli.Command{
+	Name:      "sign",
+	Usage:     "sign a message",
+	ArgsUsage: "<hexMessage>",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "from",
+			Usage: "signing address",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		api, closer, err := GetFullNodeAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+		ctx := ReqContext(cctx)
+
+		if !cctx.Args().Present() {
+			return fmt.Errorf("must specify message to sign")
+		}
+
+		var addr address.Address
+		if cctx.IsSet("from") {
+			addr, err = address.NewFromString(cctx.String("from"))
+		} else {
+			addr, err = api.WalletDefaultAddress(ctx)
```

Comment:
I'd say we should force the user to specify a from address, I can't imagine a usecase where i would confidently use the default address

Reply:
Happy to agree with this, change implemented :)